src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/Cache.java

```diff
@@ -83,58 +85,61 @@ public void put(
     {
         CacheEntry oldCacheEntry = cachedEntries.get(requestUrlHash);
         boolean expectSubscribers = request.getType() == Type.INITIAL_REQUEST ? true: oldCacheEntry.expectSubscribers();
-        DefaultCacheEntry cacheEntry = new DefaultCacheEntry(
+        CacheEntry cacheEntry = new CacheEntry(
                 this,
                 request,
                 expectSubscribers);
 
         if (cacheEntry.isIntendedForSingleUser())
         {
-            cacheEntry.cleanUp();
-            return;
+            cacheEntry.purge();
         }
-
-        if (oldCacheEntry == null)
+        else if (oldCacheEntry == null)
         {
-            cachedEntries.put(requestUrlHash, cacheEntry);
-            return;
+            updateCache(requestUrlHash, cacheEntry);
         }
         else if (oldCacheEntry.isUpdateBy(request))
         {
-            cachedEntries.put(requestUrlHash, cacheEntry);
-            oldCacheEntry.subscribersOnUpdate().forEach(subscriber ->
+            updateCache(requestUrlHash, cacheEntry);
+
+            oldCacheEntry.subscribers(subscriber ->
             {
-                if (oldCacheEntry instanceof UncommitedCacheEntry)
+                if (!this.serveRequest(
+                        cacheEntry,
+                        subscriber.getRequestHeaders(requestHeadersRO),
+                        subscriber.authScope(),
+                        subscriber))
                 {
-                    cacheEntry.subscribeToUpdate(subscriber);
-                }
-                else
-                {
-                    if (!this.serveRequest(
-                            cacheEntry,
-                            subscriber.getRequestHeaders(cachedRequestHeadersRO),
-                            subscriber.authScope(),
-                            subscriber))
-                    {
-                        final MessageConsumer acceptReply = subscriber.acceptReply();
-                        final long acceptReplyStreamId = subscriber.acceptReplyStreamId();
-                        final long acceptCorrelationId = subscriber.acceptCorrelationId();
-                        this.writer.do503AndAbort(acceptReply, acceptReplyStreamId, acceptCorrelationId);
-                    }
+                    final MessageConsumer acceptReply = subscriber.acceptReply();
+                    final long acceptReplyStreamId = subscriber.acceptReplyStreamId();
+                    final long acceptCorrelationId = subscriber.acceptCorrelationId();
+                    this.writer.do503AndAbort(acceptReply, acceptReplyStreamId, acceptCorrelationId);
                 }
             });
-            oldCacheEntry.cleanUp();
+            oldCacheEntry.purge();
         }
         else
         {
-            cacheEntry.cleanUp();
+            cacheEntry.purge();
             if (request.getType().equals(Request.Type.CACHE_REFRESH))
```

Comment:
This is an `enum` so use `==` equality.

`if (request.getType() == Request.Type.CACHE_REFRESH)` ...